### DISCUSSION TITLE
Datagrid: clear button for filters

### DISCUSSION THREAD

**Discussion by derkoe** [vmware-clarity/ng-clarity#225] [Ideas]
I would like to add a delete/clear button to the Datagrid filters. This would clear the currently set filter. 

Here is what that would look like:
![image](https://user-images.githubusercontent.com/123199/177158223-6344a371-b2b9-4869-b2d8-5699901359c2.png)

API-wise I would add a `clear` method to the `ClrDatagridFilterInterface` and implement that for string and number filter. 
```
index eda2e79c7..96c2cc2e6 100644
--- a/projects/angular/src/data/datagrid/interfaces/filter.interface.ts
+++ b/projects/angular/src/data/datagrid/interfaces/filter.interface.ts
@@ -11,6 +11,8 @@ export interface ClrDatagridFilterInterface<T, S = any> {
 
   accepts(item: T): boolean;
 
+  clear?(): void;
+
   changes: Observable<any>;
 ```

The delete/clear button is only shown when the filter supports clearing.

**Comment by derkoe** [comment]
I have created a basic implementation for this feature here: https://github.com/vmware-clarity/ng-clarity/compare/main...derkoe:feat/clear-filter-button

**Reply by d-m-s** [reply]
@gnomeontherun, @hippee-lee could you guys give us some feedback on this?

**Reply by derkoe** [reply]
Here is the PR https://github.com/vmware-clarity/ng-clarity/pull/246

**Comment by colinreedmiller** [comment]
@lil-kim Please take a look at this.

**Comment by d-m-s** [comment]
any updates to this one? @lil-kim @colinreedmiller

**Reply by lil-kim** [reply]
Design work in progress
https://www.figma.com/file/vheBPLiaaBcZu4nNuBfLYb/GH%23225-Clear-filter-feature?node-id=0%3A1&t=hbj4xi2JQr0barwZ-1
